### ISSUE TITLE
Output not cleared when using `@out.capture(clear_output=True, wait=True)`

### ISSUE THREAD

**Issue by deluksic** [jupyter-widgets/ipywidgets#2970]
Consider this code:
```python
from ipywidgets import Output

out = Output()
display(out)

@out.capture(clear_output=True)
def print_my_value(value):
    if value is not None:
        print(value)
    
print_my_value(10)
print_my_value(None)
```

As you would expect, this produces an empty cell. However, if you specify `wait=True`, the cell is never cleared. My guess is that it waits for another write to the output to clear it, but it should not wait for that write forever, but only till the end of the function call, right?

**Comment by deluksic** [comment]
Workaround for now I found to be simply calling `print()` at the end of the function, which forces it to clear if the function itself doesn't actually produce anything.

**Comment by jasongrout** [comment]
> My guess is that it waits for another write to the output to clear it

Yes, that's how it was intended. The idea here is that we want as small of a time as possible between deleting and replacing the content so the page doesn't jump.


> but it should not wait for that write forever, but only till the end of the function call, right?

Good point. I suppose for most workflows, that would make sense (i.e., until the capture context exited).

**Comment by jasongrout** [comment]
Implementation thoughts: It may be as simple as sending `print()` when the context handler exits - the key thing to check here would be if that actually generates output, like a newline, etc. That's still kind of a kludge, though. Probably more correct is to send a new message to the frontend widget to effect a clear, but only if nothing else has triggered the clear, i.e., if a clear is pending, go ahead and execute it.

**Comment by deluksic** [comment]
Yup, this is what I ended up doing. This has an unfortunate side-effect of adding an empty line, but nothing major.
